[RELAND] Remove some unnecessary <iostream> includes from headers

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #108150
* #108149

In almost all cases this is only included for writing the output formatter, which
only uses `std::ostream` so including `<ostream>` is sufficient.

The istream header is ~1000 lines so the difference is non-trivial.

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @mcarilli @ptrblck @leslie-fang-intel @EikanWang @voznesenskym @penguinwu @Guobing-Chen @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng @anijain2305

peterbell10 committed:
Remove some unnecessary <iostream> includes from headers

In almost all cases this is only included for writing the output formatter, which
only uses `std::ostream` so including `<ostream>` is sufficient.

The istream header is ~1000 lines so the difference is non-trivial.

[ghstack-poisoned]

albanD commented on .github/ci_commit_pins/xla.txt:
```diff
@@ -1 +1 @@
-709c75a04d9b35d83a9509e1534a8aa2046b8912
+c3c16ccac41cb2db6ba88fb31342f4af62c7e15a
```
Should not be there?

peterbell10 replied:
This is for https://github.com/pytorch/xla/pull/5434, xla was relying on indirectly including iostream through our headers.

albanD reviewed (approved):
Ok!

albanD commented:
@pytorchbot merge -i

albanD commented:
@pytorchbot merge -r

albanD commented:
@peterbell10 could you fix the merge conflicts please?

peterbell10 committed:
Update on "[RELAND] Remove some unnecessary <iostream> includes from headers"


In almost all cases this is only included for writing the output formatter, which
only uses `std::ostream` so including `<ostream>` is sufficient.

The istream header is ~1000 lines so the difference is non-trivial.

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 mcarilli ptrblck leslie-fang-intel EikanWang voznesenskym penguinwu Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng anijain2305

[ghstack-poisoned]

peterbell10 commented:
@pytorchbot merge -i
